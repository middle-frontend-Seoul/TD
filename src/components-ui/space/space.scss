.space {
  display: flex;
  min-width: 100%;
  justify-content: space-between;
}
.space_type-horizontal {
  flex-direction: row;
  margin-left: -12px;
  margin-right: -12px;
}
.space_type-vertikale {
  flex-direction: column;
  margin-top: -12px;
  margin-bottom: -12px;
}
.space__horizontal_size-medium {
  margin-left: 12px;
  margin-right: 12px;
}
.space__vertikale_size-medium {
  margin-top: 12px;
  margin-bottom: 12px;
}
