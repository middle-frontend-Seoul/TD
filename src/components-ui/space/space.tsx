import React, { FC } from 'react';
import cn from 'classnames';

import './space.scss';

export type SpaceType = 'horizontal' | 'vertical';

export type SpaceSize = 'medium';

export interface IBlockProps {
  size?: SpaceSize;
  type?: SpaceType;
  position?: 'center';
  className?: string;
  children: React.ReactNode;
}

export const Space: FC<IBlockProps> = ({
  type = 'vertical',
  size = 'medium',
  position,
  className,
  children,
}) => {
  const classes = cn(className, 'space', `space_type-${type}`, {
    position: Boolean(position),
  });

  const childs = React.useMemo(() => {
    return React.Children.toArray(children)
      .map((child, i) => {
        if (!React.isValidElement(child)) {
          return false;
        }

        return React.cloneElement(child, {
          key: String(i),
          className: `space__${type}_size-${size}`,
        });
      })
      .filter(Boolean);
  }, [children, type, size]);

  return <div className={classes}>{childs}</div>;
};
