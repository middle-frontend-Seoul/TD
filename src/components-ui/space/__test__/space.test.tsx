import React from 'react';
import renderer from 'react-test-renderer';

import { Space } from '../space';

it('renders space vertical', () => {
  const tree = renderer
    .create(
      <Space>
        <div>1</div>
        <div>1</div>
      </Space>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders space horizontal', () => {
  const tree = renderer
    .create(
      <Space type="horizontal">
        <div>1</div>
        <div>1</div>
      </Space>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
