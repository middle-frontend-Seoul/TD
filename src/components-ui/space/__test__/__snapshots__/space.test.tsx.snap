// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders space horizontal 1`] = `
<div
  className="space space_type-horizontal"
>
  <div
    className="space__horizontal_size-medium"
  >
    <div>
      1
    </div>
  </div>
  <div
    className="space__horizontal_size-medium"
  >
    <div>
      1
    </div>
  </div>
</div>
`;

exports[`renders space vertical 1`] = `
<div
  className="space space_type-vertical"
>
  <div
    className="space__vertical_size-medium"
  >
    <div>
      1
    </div>
  </div>
  <div
    className="space__vertical_size-medium"
  >
    <div>
      1
    </div>
  </div>
</div>
`;
